src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Query/PropertyPages/CategoryFromRuleDataProducer.cs

```diff
@@ -9,19 +9,20 @@
 namespace Microsoft.VisualStudio.ProjectSystem.VS.Query
 {
     /// <summary>
-    /// Handles retrieving a set of <see cref="ICategory"/>s from an <see cref="IPropertyPage"/>.
+    /// Handles retrieving a set of <see cref="ICategory"/>s from an <see cref="IPropertyPage"/>
+    /// or <see cref="ILaunchProfile"/>.
     /// </summary>
-    internal class CategoryFromPropertyPageDataProducer : QueryDataFromProviderStateProducerBase<PropertyPageProviderState>
+    internal class CategoryFromRuleDataProducer : QueryDataFromProviderStateProducerBase<ContextAndRuleProviderState>
```

Comment:
Just a Q. What is with the 'From' naming semantic? Wouldn't `RuleCategoryDataProducer` work just as well? Before, it would have been `PropertyPageCategoryDataProducer` which also works.

Reply:
I wanted to emphasize that the category was being produced from the `Rule`--that is, there's a hierarchy in play here.